### ISSUE TITLE
`yarn` is erased after `global add`

### ISSUE THREAD

**Issue by selfagency** [yarnpkg/yarn#7937]
### Bug description
After installing a package globally, the `yarn` executable is erased from `/usr/local/bin`, requiring reinstallation. I have been experiencing this consistently across three different Macs.

**Command**

```sh
yarn global add vue-language-server@0.0.66
```

**What is the current behavior?**
After installing a package globally, the `yarn` executable is deleted from /usr/local/bin.

![Capto_Capture 2020-02-26_02-47-49_PM 2020-02-26 14_53_32](https://user-images.githubusercontent.com/2541728/75382112-e1d77d00-58a7-11ea-8607-1140b47a54a7.gif)

**What is the expected behavior?**
For `yarn` not to be erased.

**Steps to Reproduce**

1. Install `yarn` with Homebrew.
2. Run `yarn global add [whatever]`

**Environment**

- Node Version: `10.18.0`
- Yarn v1 Version: `1.22.0` 
- OS and version: Mac OS 10.15.3

**Comment by seognil** [comment]
same here :(

**Comment by selfagency** [comment]
Well at the very least, I've discovered you can just run `brew unlink yarn; brew link yarn;` instead of having to reinstall it every time this happens.

**Comment by seognil** [comment]
@selfagency 
I found a tricky solution, no need to `brew link` every time.

First, make two symbolic links manually (at some where)
```
ln -s /usr/local/lib/node_modules/npm/bin/npm-cli.js /usr/local/opt/node/bin/npm
ln -s /usr/local/lib/node_modules/npm/bin/npm-cli.js /usr/local/opt/node/bin/npx
```

Then add the folder of links we just created to `~/.bashrc` or `~/.zshrc`. As we explicitly control the executable path, so it would always found `npm` and `yarn` no matter how `/usr/local/bin` is messed up by the issue.
```
export PATH="/usr/local/opt/node/bin:$PATH"
export PATH="/usr/local/opt/yarn/bin:$PATH"
```

(Just as how `brew install node@12` said, the idea is the same)
